tests/ssl-opt.sh

```diff
@@ -1860,6 +1860,48 @@ run_test    "Opaque key for server authentication: invalid alg: TLS-ECDHE-ECDSA
             -s "error" \
             -c "error"
 
+requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_2
+requires_config_enabled MBEDTLS_USE_PSA_CRYPTO
+requires_config_enabled MBEDTLS_X509_CRT_PARSE_C
+requires_config_enabled MBEDTLS_ECDSA_C
+requires_config_enabled MBEDTLS_SHA256_C
+run_test    "Two Opaque keys for server authentication: EC with different algs" \
+            "$P_SRV auth_mode=required key_opaque=1 crt_file=data_files/server5.crt \
+             key_file=data_files/server5.key key_opaque_algs=ecdh,none \
```

Comment:
It would be nice to checked that the right key has been selected. I think this can be done by using different certs then checking which cert was used from the output. (See `tests/data_files/Readme-x509.txt` for list of certs, can't guarantee it's up-to-date but it's a start.) For example one could use server5.key on one alg and server6.key with the other, and then check in the output whether the serial number is 9 or 10.

Also, ideally I'd like a pair of tests, both with the same key setup on the server: one where the client forces an ECDHE-ECDSA suite and we select the key that can sign, and the other forcing ECDH-ECDSA and we select the key that can do ECDH.